### PR TITLE
feat: add a form for API key validation

### PR DESCRIPTION
**Pull request by st-jay**

### CODE REVIEW

File: src/images.ts [hunk 1]
```diff
@@ -3,4 +3,5 @@
  * 
  * Note: Never use local imports or relative paths when using images in src etc
  */
-import "./logo.svg";
\ No newline at end of file
+import "./logo.svg";
+import "./assets/form-field-error-icon.png";
```
Comment (nkshah2) [inline src/images.ts:7]: Use svgs where possible

File: config/webpack.config.js [hunk 2]
```diff
@@ -366,11 +366,10 @@ module.exports = function (webpackEnv) {
...
               type: 'asset',
-              parser: {
-                dataUrlCondition: {
-                  maxSize: imageInlineSizeLimit,
-                },
-              },
+              loader: require.resolve('file-loader'),
+              options: {
+                name: 'static/media/[name].[ext]'
+              }
```
Comment (nkshah2) [inline config/webpack.config.js:372]: I dont think this will be required if you use svgs and not a png

Comment (nkshah2) [reply config/webpack.config.js:372]: Also we could just increase the imageInlineSizeLimit to accommodate this no?

File: package-lock.json [hunk 3]
```diff
@@ -5,7 +5,6 @@
   "requires": true,
   "packages": {
     "": {
-      "name": "dashboard",
```
Comment (nkshah2) [inline package-lock.json:8]: Why was this removed?

File: src/ui/components/button/Button.module.css [hunk 4]
```diff
@@ -0,0 +1,16 @@
+.button {
+  padding: 10px 16px;
+  font-size: 14px;
+  line-height: 17px;
+  background-color: rgb(255, 153, 51);
+  border: 1px solid rgb(255, 128, 0);
+  font-weight: 600;
+  color: white;
+  border-radius: 6px;
+  font-family: 'Rubik';
```
Comment (nkshah2) [inline src/ui/components/button/Button.module.css:10]: Its easier to just apply Rubik to the body itself, then individually handle cases that are not rubik

File: src/ui/components/button/Button.module.css [hunk 5]
```diff
@@ -0,0 +1,16 @@
+.button {
+  padding: 10px 16px;
+  font-size: 14px;
+  line-height: 17px;
+  background-color: rgb(255, 153, 51);
+  border: 1px solid rgb(255, 128, 0);
+  font-weight: 600;
+  color: white;
```
Comment (nkshah2) [inline src/ui/components/button/Button.module.css:8]: Use css variables (and dont just call it white), this project should never use colors by names or hex codes directly

File: src/ui/components/button/Button.module.css [hunk 6]
```diff
@@ -0,0 +1,16 @@
+.button {
+  padding: 10px 16px;
+  font-size: 14px;
+  line-height: 17px;
+  background-color: rgb(255, 153, 51);
+  border: 1px solid rgb(255, 128, 0);
+  font-weight: 600;
```
Comment (nkshah2) [inline src/ui/components/button/Button.module.css:7]: Use variables

File: src/ui/components/button/Button.module.css [hunk 7]
```diff
@@ -0,0 +1,16 @@
+.button {
+  padding: 10px 16px;
+  font-size: 14px;
```
Comment (nkshah2) [inline src/ui/components/button/Button.module.css:3]: Use variables (basically it should be dependent on the design guidelines, if the titles are supposed to be 20px then that should be a --font-size-title variable and so on

File: src/ui/components/button/Button.module.css [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+  background-color: rgb(255, 153, 51);
+  border: 1px solid rgb(255, 128, 0);
+  font-weight: 600;
+  color: white;
+  border-radius: 6px;
+  font-family: 'Rubik';
+  cursor: pointer;
+}
+
+.button[data-full-width="true"] {
```
Comment (nkshah2) [inline src/ui/components/button/Button.module.css:14]: Why not just have a class button-full-width?

File: src/ui/components/button/Button.module.css [hunk 9]
```diff
@@ -0,0 +1,16 @@
+.button {
```
Comment (nkshah2) [inline src/ui/components/button/Button.module.css:1]: Have you tested how bundling works in dev and production mode? Does it all get bundled into a single css file in production mode?

File: src/ui/components/button/Button.tsx [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+  type: "submit";
+  disabled?: boolean;
+  fullWidth?: boolean;
+  handleClick?: React.MouseEventHandler<HTMLButtonElement>;
+};
+
+const Button: React.FC<ButtonPropType> = (props) => {
+  return (
+    <button
+      className={styles.button}
```
Comment (nkshah2) [inline src/ui/components/button/Button.tsx:30]: Dont need to do it this way, look at the way its done in App.tsx

```tsx
import './App.css';

...

className="App-logo"
```

File: src/ui/components/button/Button.tsx [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+      onClick={props.handleClick}
+      type={props.type}
+      disabled={props.disabled}
+      data-full-width={props.fullWidth}
```
Comment (nkshah2) [inline src/ui/components/button/Button.tsx:34]: Everything is coming from props, whats the point of having this component then?

File: src/ui/components/inputField/InputField.module.css [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+.inputFieldError {
+  display: flex;
+  align-items: center;
+  margin-top: 8px;
+}
+
+.inputFieldErrorIcon {
+  height: 15px;
```
Comment (nkshah2) [inline src/ui/components/inputField/InputField.module.css:47]: This is just me, but I prefer dimensions for spacing and layout be multiples of 4 or 8, makes the overall layout cleaner

File: src/ui/pages/auth/Auth.tsx [hunk 13]
```diff
@@ -12,66 +12,95 @@
...
         });
 
         const body = await response.json();
 
         if (response.status === 200 && body.status === "OK") {
-            localStorageHandler.setItem(StorageKeys.API_KEY, apikeyFromUser);
+            localStorageHandler.setItem(StorageKeys.API_KEY, apiKey);
             navigate("/");
+        } else if (response.status === UNAUTHORISED_STATUS) {
+            setApiKeyFieldError("Please enter a valid API Key.")
```
Comment (nkshah2) [inline src/ui/pages/auth/Auth.tsx:50]: "Invalid API Key"

File: src/ui/pages/auth/Auth.tsx [hunk 14]
```diff
@@ -12,66 +12,95 @@
...
         } else {
-            didPrompt = false;
-            promptAndValidateKey();
+            setApiKeyFieldError("An error occured while validating API Key.");
```
Comment (nkshah2) [inline src/ui/pages/auth/Auth.tsx:52]: "Something went wrong"

File: src/ui/pages/auth/Auth.tsx [hunk 15]
```diff
@@ -12,66 +12,95 @@
...
+        const { value } = e.target;
+        setApiKey(value);
+        setApiKeyFieldError("");
+    }
+
     return (
-        <div>Loading...</div>
+        <div className={styles.pageContainer}>
+            <div className={styles.apiKeyFormContainer}>
+                <h1 className={styles.apiKeyFormTitle}>Validate your API Key</h1>
```
Comment (nkshah2) [inline src/ui/pages/auth/Auth.tsx:83]: "Enter your API key"

Comment (nkshah2) [reply src/ui/pages/auth/Auth.tsx:83]: Also remove the divider under the label, looks strange

File: src/ui/pages/auth/Auth.module.css [hunk 16]
```diff
@@ -0,0 +1,30 @@
...
+  align-items: center;
+}
+
+.apiKeyFormContainer {
+  display: flex;
+  flex-flow: column;
+  padding: 40px;
+  border-radius: 12px;
+  box-shadow: 0px 3px 6px rgba(0, 0, 0, 0.16);
+  width: 476px;
```
Comment (nkshah2) [inline src/ui/pages/auth/Auth.module.css:16]: Use percentages or add responsive css

File: src/utils.ts [hunk 17]
```diff
@@ -78,4 +78,42 @@ export const fetchDataAndRedirectIf401 = async ({
     }
 
     return response;
-}
\ No newline at end of file
+}
+
+export const fetchData = async ({
```
Comment (nkshah2) [inline src/utils.ts:83]: Why is this function needed?

**Review by nkshah2 — APPROVED** [review]

File: config/webpack.config.js [hunk 18]
```diff
@@ -366,11 +366,10 @@ module.exports = function (webpackEnv) {
             {
```
Comment (nkshah2) [inline config/webpack.config.js:366]: Commenting to request changes

**Review by nkshah2 — CHANGES_REQUESTED** [review]

File: src/utils.ts (hunk 17 above)
Comment (nkshah2) [reply src/utils.ts:83]: Change the above function to call this one internally and handle 401

**Review by nkshah2 — CHANGES_REQUESTED** [review]

File: src/images.ts (hunk 1 above)
Comment (st-jay) [reply src/images.ts]: Updated

File: config/webpack.config.js (hunk 2 above)
Comment (st-jay) [reply config/webpack.config.js]: removed the changes for now as no PNGs are used

File: package-lock.json (hunk 3 above)
Comment (st-jay) [reply package-lock.json]: Updated ( it was removed due to my npm version being `7.5` )

File: src/ui/components/button/Button.module.css (hunk 4 above)
Comment (st-jay) [reply src/ui/components/button/Button.module.css]: Updated

File: src/ui/pages/auth/Auth.tsx (hunk 13 above)
Comment (st-jay) [reply src/ui/pages/auth/Auth.tsx]: Updated

File: src/ui/pages/auth/Auth.tsx (hunk 14 above)
Comment (st-jay) [reply src/ui/pages/auth/Auth.tsx]: Updated

File: src/ui/pages/auth/Auth.tsx (hunk 15 above)
Comment (st-jay) [reply src/ui/pages/auth/Auth.tsx]: Updated

**Review by nkshah2 — CHANGES_REQUESTED** [review]

File: src/ui/components/inputField/InputField.css [hunk 19]
```diff
@@ -0,0 +1,45 @@
+.input-field-container {
+  display: flex;
+  flex-flow: column;
+  margin-bottom: 24px;
+}
+
+.input-label {
+  color: rgb(110, 106, 101);
+  font-weight: 500;
+  margin-bottom: 14px;
```
[the remark this answers is not in the document]
Comment (st-jay) [reply src/ui/components/inputField/InputField.css]: Updated wherever it was possible

**Review by nkshah2 — APPROVED** [review]
